event-handler/teleport_events_watcher.go

```diff
@@ -282,3 +283,30 @@ func (t *TeleportEventsWatcher) Events(ctx context.Context) (chan *TeleportEvent
 func (t *TeleportEventsWatcher) StreamSessionEvents(ctx context.Context, id string, index int64) (chan events.AuditEvent, chan error) {
 	return t.client.StreamSessionEvents(ctx, id, index)
 }
+
+// UpsertLock upserts user lock
+func (t *TeleportEventsWatcher) UpsertLock(ctx context.Context, user string, login string, period time.Duration) error {
+	var expires *time.Time
+
+	if period > 0 {
```

Comment:
So the user is locked forever if no period is set, yes? At that point is revoking the lock a job for the Teleport admin?

Reply:
And if someone configures a negative period, does the lock immediately expire? Is it worth validating that on load?

Reply:
1. Yes, that is correct. That was the original requirement: lock forever by default.
2. In this case, negative period would be equal to LockEnable: false. If one have access to setting Period, they also could change Enabled. I think, it's fine as is. I also think, that API would fail if the period is negative.